Add VectorField and VectorFieldRenderer to enable multiple VFs in one View

GPMueller opened the pull request:
### **This PR is so far only meant for discussion of this refactoring I tried**
*The demo has not been updated yet.*


- The `VectorField` class contains all vector-field-specific data previously contained in `View`. It also provides functions to update the geometry, directions, or both.
- The `VectorFieldRenderer` is an abstraction to distinguish from `Renderer`s which do not contain a VF, such as `CombinedRenderer` or `CoordinateSystemRenderer`.
- Also added the Bounded camera rotation type


**Questions:**
- *does this clutter the class structure too much?*
- *does this complicate the simple use cases too much?*


**My Opinions:**
- *it is more intuitive to have multiple vector-fields than to **need** multiple `View`s*
- *it makes sense to use multiple `View`s when e.g. different camera settings for the different objects are wanted for some reason*
- *this gives a clear hierachy of `Renderer`s, where only some actually visualise a vector-field*
- *this way it is easy to create complex combinations of different vector-fields without needing wrapper functions to handle synchronization of `Option`s*

FlorianRhiem commented:
- Each individual renderer should only be used in a single `View`.
    Using a renderer in two or more `Views` which share the same OpenGL context will work, however not all `Views` have to be inside same context. Previously, this was ensured by using the `View` reference as part of the `RendererBase` constructor.
- What benefit is gained by having to seperate between `VectorFieldRenderers` and other renderers?
    Previously, you could call `updateIfNecessary()` on all renderers and only those which needed the geometry / vectors information used it. The user did not need to care, with this pull request they do.
    The only benefit that I see is that since you pass the `VectorField` by reference, you would need a placeholder `VectorField` in some rare cases (e.g. a `CombinedRenderer` without any renderers). If this was the reason for introducing `VectorFieldRenderer` I would recommend moving its interface to `RendererBase` so that the user can treat all renderers equally, but does not need any placeholder `VectorField`.
- The camera and lighting changes are unrelated and should not be part of this pull request / commit.

GPMueller commented:
1. I do not see why a `Renderer` cannot be used across OpenGL contexts, but I will try to simply replace the `const Options& options` in the constructors with `const View& view` again
2.  - the benefit is that I gain flexibility without adding much complexity (I would even argue that passing also a `Vectorfield &` to a `VectorfieldRenderer` is a marginal change).
    - Why would the `updateIfNecessary()` be changed? The logic there stays the same.
    - What should the `CombinedRenderer` do when it's renderers contain different `VectorField`s - which should it then pass on to the `RendererBase`?
3. How should the `VectorFieldRenderer` be moved into `RendererBase` without requiring a placeholder? How would the user set the data for e.g. `GlyphRenderer`, while not being able to set data for e.g. `CombinedRenderer`?
4. Right. I will remove camera and lighting from this PR.

GPMueller committed:
Add VectorField and VectorFieldRenderer to enable multiple VFs in one View.
- The `VectorField` class contains all vector-field-specific data previously contained in  `View`. It also provides functions to update the geometry, directions, or both.
- The `VectorFieldRenderer` is an abstraction to distinguish from `Renderer`s which do not contain a VF, such as `CombinedRenderer` or `CoordinateSystemRenderer`.

GPMueller commented:
Squashed the commits into one with f1e675b.
The changes should now be ready to be merged (demos and Makefile have also been updated).

GPMueller commented:
**TODO:**
- [x] Update Readme
- [x] Add non-default constructor to VectorField

GPMueller committed:
Changed VectorField constructor to non-default.

GPMueller committed:
Updated Readme with info on the new VectorField.
